examples/lib/libcapture/main.c

```diff
@@ -0,0 +1,155 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata.h"
+#include "runmodes.h"
+#include "conf.h"
+#include "pcap.h"
+#include "runmode-lib.h"
+#include "source-lib.h"
+#include "threadvars.h"
+
+static int worker_id = 1;
+
+/* Suricata worker thread in library mode.
+   The functions should be wrapped in an API layer. */
+static void *SimpleWorker(void *arg)
+{
+    char *pcap_file = (char *)arg;
+
+    /* Create worker. */
+    ThreadVars *tv = SCRunModeLibCreateThreadVars(worker_id++);
+    if (!tv) {
+        pthread_exit(NULL);
+    }
+
+    /* Start worker. */
+    if (SCRunModeLibSpawnWorker(tv) != 0) {
+        pthread_exit(NULL);
+    }
+
+    /* Replay pcap. */
+    pcap_t *fp = pcap_open_offline(pcap_file, NULL);
+    if (fp == NULL) {
+        pthread_exit(NULL);
+    }
+
+    LiveDevice *device = LiveGetDevice("lib0");
+    assert(device != NULL);
+
+    int datalink = pcap_datalink(fp);
+    int count = 0;
+    struct pcap_pkthdr pkthdr;
+    const u_char *packet;
+    while ((packet = pcap_next(fp, &pkthdr)) != NULL) {
+        if (TmModuleLibHandlePacket(tv, device, packet, datalink, pkthdr.ts, pkthdr.len, 0, 0) !=
+                0) {
+            pthread_exit(NULL);
+        }
```

Comment:
I'm not thinking that `TmModuleLibHandlePacket` and the call to it should be changed to something like:
- Some helper function to get a packet and perhaps some initial setup. But allow the user to to do some setup as well, like a "release" function.
- User then calls a `TmThreadSlowProcessPkt` or some wrapper around it.
This removes one function from `source-lib.c` making it more functional, as well as allows post-inspection of the packet (which would make sense in workers mode, but not autofp)

Reply:
Yes not a bad idea.
Will there be an equivalent `TmThreadFastProcessPkt` to `TmThreadSlowProcessPkt`? And what would the difference be?

Reply:
Oops, typo.. Meant `TmThreadsSlotProcessPkt`.